parser.js

```diff
@@ -497,70 +496,98 @@ class DataflashParser {
+                // For each field
+                for (let j = 0; j < num_fields; j++) {
+                    if (j == time_index) {
+                        // us to ms
+                        parsed.time_boot_ms[i] = this.parse_type(Format.charAt(j)) / 1000.0
+                    } else {
+                        const name = msg.Columns[j]
+                        parsed[name][i] = this.parse_type(Format.charAt(j))
+                    }
+                }
+
+                if (i % 1000 === 0) {
+                    const perc = 100 * i / len
+                    self.postMessage({ percentage: perc })
+                }
             }
+
+            return parsed
         }
+
         delete this.messages[name]
-        this.messages[name] = parsed
 
-        self.postMessage({ percentage: 100 })
-        console.log(name, this.messageHasInstances(name) ? 'has instances' : 'has no instances')
-        if (parsed.length && this.messageHasInstances(name)) {
-            const instanceField = this.getInstancesFieldName(name)
-            const instances = {}
-            for (const msg of parsed) {
-                try {
-                    instances[msg[instanceField]].push(msg)
-                } catch (e) {
-                    instances[msg[instanceField]] = [msg]
-                }
-            }
-            if (Object.keys(instances).length === 1) {
-                this.fixDataOnce(name)
-                this.simplifyData(name)
-                this.postData({ messageType: name, messageList: this.messages[name] })
-                return parsed
+        const has_instance = this.messageHasInstances(name)
+        console.log(name, has_instance ? 'has instances' : 'has no instances')
+
+        if (has_instance && ("InstancesOffsetArray" in msg_FMT) && (Object.keys(msg_FMT.InstancesOffsetArray).length > 1)) {
+            // Parse instances
+            for (const [index, offsets] of Object.entries(msg_FMT.InstancesOffsetArray)) {
+                const inst_name = name + '[' + index + ']'
+                this.messages[inst_name] = parse(msg_FMT, offsets)
+                this.postData({ messageType: inst_name, messageList: this.messages[inst_name] })
             }
-            for (const [index, messages] of Object.entries(instances)) {
-                const newName = name + '[' + index + ']'
-                this.messages[newName] = messages
-                this.fixDataOnce(newName)
-                this.simplifyData(newName)
-                this.postData({
-                    messageType: newName,
-                    messageList: this.messages[newName]
-                })
+            self.postMessage({ percentage: 100 })
+
+
+            // Old behavior was to return the un-fixed and un-split data here
+            // Putting something in this array makes WebTools work but its dumb
+            // Could be anything....
+            this.messages[name] = { found_log: true }
+            return this.messages[name]
```

Comment:
This is the key change in behavior, we used to populate the base message as well as the instances. Although we would only do the clean up on the instances.

To get WebTools to work in its current form I need something in the base message if it was found. That is because I was using this pattern to tell if we got data:

https://github.com/ArduPilot/WebTools/blob/2cd7e3421b24a6cd547dda1125769e984e35ca19/HardwareReport/HardwareReport.js#L2136